### PR TITLE
Change gptransfer to check matching tablespaces

### PR DESCRIPTION
**Pull request by jmcatamney**
The previous check in gptransfer for matching source and destination filespaces
has been changed to a check for tablespaces, since filespaces no longer exist.

### CODE REVIEW

**Comment by xinzweb** [comment]
+1

File: gpMgmt/bin/gptransfer [hunk 1]
```diff
@@ -3332,48 +3332,48 @@ class GpTransfer(object):
...
             # For transferring in full mode
-            source_fs_sql = """SELECT fsname FROM pg_catalog.pg_filespace"""
-            self._check_filespace(source_host, source_port, 'template1', source_user, list_dest_fs, source_fs_sql)
+            source_ts_sql = """SELECT spcname FROM pg_catalog.pg_tablespace"""
+            self._check_tablespace(source_host, source_port, 'template1', source_user, list_dest_ts, source_ts_sql)
         else:
             # For transferring in table mode
-            template_source_fs_sql = """SELECT fsname FROM pg_catalog.pg_filespace WHERE oid IN
-                                        (SELECT spcfsoid FROM pg_catalog.pg_tablespace
+            template_source_ts_sql = """SELECT spcname FROM pg_catalog.pg_tablespace WHERE oid IN
```
Comment (jimmyyih) [inline gpMgmt/bin/gptransfer]: This query looks weird.  Should it look something like this:
```
SELECT spcname FROM pg_catalog.pg_tablespace WHERE oid IN
(SELECT dattablespace FROM pg_catalog.pg_database WHERE datname = '%s')
OR oid IN
(SELECT reltablespace FROM pg_catalog.pg_class WHERE relname = '%s');
```

Comment (jmcatamney) [reply gpMgmt/bin/gptransfer]: If it's more readable that way, I can change it.

Comment (jimmyyih) [reply gpMgmt/bin/gptransfer]: It's not the readability.  It's that there was an extra unnecessary query to pg_tablespace.

File: gpMgmt/test/behave/mgmt_utils/steps/data/gptransfer_setup/filespace_tablespace.sql [hunk 2]
```diff
@@ -1,7 +1,4 @@
--- This require a filespace 'fs' created before running this test
--- refer to the behave test case which creates the filespace
-
-CREATE TABLESPACE ts filespace fs;
+CREATE TABLESPACE ts;
```
Comment (jimmyyih) [inline gpMgmt/test/behave/mgmt_utils/steps/data/gptransfer_setup/filespace_tablespace.sql]: This line would error out without specifying the on-disk location:
```
postgres=# create tablespace ts;
ERROR:  syntax error at or near ";"
LINE 1: create tablespace ts;
                            ^
postgres=# create tablespace ts location '/tmp/foobar';
CREATE TABLESPACE
```

Comment (jmcatamney) [reply gpMgmt/test/behave/mgmt_utils/steps/data/gptransfer_setup/filespace_tablespace.sql]: Fixed, thanks.

**Pull request merged**